build.proj

```diff
@@ -3,9 +3,9 @@
   <Import Project="Directory.Build.props" />
 
   <PropertyGroup>
-    <BuildNative Condition="'$(BuildNative)' == ''">true</BuildNative>
+    <BuildNative Condition="'$(BuildNative)' == '' and '$(DirectoryToBuild)' == ''">true</BuildNative>
```

Comment:
Observation: DirectoryToBuild appears  14 times in this commit.  Wish we didn't need to special case something that's only excercised through manual interaction so much.

Reply:
Also, where is DirectoryToBuild documented along with how it's supposed to be used?  What real value does it provide over someone just directly building a project?

Reply:
We need to document DirectoryToBuild properly I think. The biggest advantage of it is that it allows building a leaf project without the need to invoke msbuild manually. This is very convenient on machines where you don't have a fitting global SDK installed and don't want to set env vars to point to the local one.

Reply:
It just seems rather odd to me that the only thing you can specify for DirectoryToBuild is something like src\foo, there are odd assumptions about the substructure of the library folders that aren't really all that useful.  Added to that the footprint of this feature across a bunch of different files and it led me to ask this question about how its supposed to work and what's the real value.